pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -177,13 +202,18 @@ func (h *handler) deployK3sBasedUpgradeController(clusterName string, isK3s, isR
 			}
 		}
 
-		if app.Spec.ExternalID == latestVersionID {
-			// app is up to date, no action needed
+		desiredApp := app.DeepCopy()
+		externalIDIsCorrect := app.Spec.ExternalID == latestVersionID
+		pspValuesHaveBeenSet := desiredApp.Spec.ValuesYaml == fmt.Sprintf(ConditionalPSPValues, pspEnabledValue)
```

Comment:
We should move `desiredApp := app.DeepCopy()` to directly before `desiredApp` get modified, since it looks like we perform the copy but may not potentially need it.

Reply:
Good catch, updated